### PR TITLE
feat: watch video and copy link

### PR DESCRIPTION
**Pull request by krgarima**
##Change Logs:

#Description: Added Watch video and Copy video link feature

## More details:
1. User will be redirected to a single video page where he can watch the video
2. Added buttons (only UI - Like, watch later, copy and save to playlist)
3. User can copy the link of the video

What not to check?
- Responsiveness
- Using css variables

Screenshots:
av6

### CODE REVIEW

**Review by athar05 — COMMENTED** [review]
The code looks clean. Try to add variables wherever possible.

**Comment by athar05** [comment]
The UI looks good. I think you can merge it.

File: src/components/SingleVideo/SingleVideo.js [hunk 1]
```diff
@@ -1,5 +1,50 @@
...
-  return <div>SingleVideo</div>;
+  const { videoList } = useContext(videoListContext);
+
+  const videoId = useParams();
+  let location = useLocation();
+
+  const findVideo = videoList.find((video) => video._id === videoId.videoId);
+  const { title, creator, genre, date, views, url } = findVideo;
+
+  const encodedToken = localStorage.getItem("token");
```
Comment (vishalg8454) [inline src/components/SingleVideo/SingleVideo.js:15]: Should not side effects be inside `useEffect()` ?

**Review by vishalg8454 — COMMENTED** [review]
All looks good, just check the comment once.

File: src/components/SingleVideo/SingleVideo.css [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+  height: 12rem;
+  width: 20rem;
+  overflow-y: scroll;
+  scrollbar-width: thin;
+  position: absolute;
+  top: 32%;
+  left: 32%;
+  padding: 5px;
+  border: 2px solid white;
+  /* background-color: #262626; */
```
Comment (chanchal-panpaliya) [inline src/components/SingleVideo/SingleVideo.css]: remove comment

File: src/components/SingleVideo/SingleVideo.css [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+  height: 2rem;
+  width: 90%;
+  margin: auto;
+}
+
+.createPlaylist button {
+  height: 2rem;
+  width: 80%;
+  margin: auto;
+  background-color: #262626;
```
Comment (chanchal-panpaliya) [inline src/components/SingleVideo/SingleVideo.css:132]: use var

**Review by chanchal-panpaliya — APPROVED** [review]
code looks clean with proper naming with good UI

**Pull request merged**